File: papi/lista3/exercicio4.py
while True:
    try:
        numero = int(input("Digite um número inteiro maior ou igual a zero para saber seu fatorial: "))
        if numero < 0:
            raise ValueError
        break
    except ValueError:
        print("Por favor digite um número válido maior que zero")
    
if numero < 2:
    print("O fatorial de", numero, "é 1")
else:
    contador = 1
    fatorial = 1
    while contador <= numero:
        fatorial *= contador
        contador +=1

    print("O fatorial de", numero, "é", fatorial)

File: papi/lista3/exercicio4.kt
package lista3

fun main() {
    var numero: Int 

    while (true) {
        try {
            print("Por favor digite um número positivo ou zero para ver seu fatorial: ")
            val input = readLine()
            numero = input!!.toInt()
            if (numero < 0) {
                throw Exception("Número inválido")
            }
            break
        } catch (e: Exception) {
            print("Por favor digite um número inteiro válido positivo ou zero")
        }
    }

    var cont = 1
    var total = 1

    if (numero < 2) {
        print("O fatorial de $numero é 1 \n")
    } else {
        while (cont <= numero) {
            total = total * cont
            cont += 1
        }
        print("O fatorial do número é $total \n")
    }
} 